// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package transaction;

// The Transaction service definition.
service Transaction {
  rpc GetTransaction (TransactionRequest) returns (TransactionResponse) {}
  rpc UpdateTransaction (UpdateTransactionRequest) returns (TransactionResponse) {}
  rpc CreateTransaction (CreateTransactionRequest) returns (TransactionResponse) {}
  rpc DeleteTransaction (UpdateTransactionRequest) returns (DeleteResponse) {}
}
message TransactionRequest {
  string transactionId = 1;
}
message CreateTransactionRequest{
  float amount = 1;
  string accountId = 2;
}

message UpdateTransactionRequest {
  string transactionId = 1;
  float amount = 2;
}

message TransactionResponse {
  string transactionId = 1;
  float amount = 2;
  string accountId = 3;
  string timestamp = 4;
}

message DeleteResponse{
}